### ISSUE TITLE
Mouseless Navigation In Database Opening Screen Does Not Work As Expected

### ISSUE THREAD

**Issue by Renegade-Master** [keepassxreboot/keepassxc#10506]
## Overview
During the opening screen (after a Database has been selected), <kbd>Tab</kbd> navigation used to allow for selecting the Hardware Key, then the Keyfile, and finally selecting the 'Open' button. This allowed for mouseless opening of a Database.

## Steps to Reproduce

1. Select a Database to open.
2. Enter the Database password.
3. Press <kbd>Tab</kbd> followed by <kbd>Space</kbd> to enable a Hardware Key.
4. Press <kbd>Tab</kbd> to navigate to Hardware Key refresh button.
5. Press <kbd>Tab</kbd> to navigate to 'I have a keyfile' button.
6. It it not possible to select the 'I have a keyfile' button without using the mouse at this point.

## Expected Behavior
1. Select a Database to open.
2. Enter the Database password.
3. Press <kbd>Tab</kbd> followed by <kbd>Space</kbd> to enable a Hardware Key.
4. Press <kbd>Tab</kbd> to navigate to Hardware Key refresh button.
5. Press <kbd>Tab</kbd> to navigate to 'I have a keyfile' button.
6. Press <kbd>Space</kbd> to open the keyfile selection window (same as Database selection window).

## Actual Behavior
1. Select a Database to open.
2. Enter the Database password.
3. Press <kbd>Tab</kbd> followed by <kbd>Space</kbd> to enable a Hardware Key.
4. Press <kbd>Tab</kbd> to navigate to Hardware Key refresh button.
5. Press <kbd>Tab</kbd> to navigate to 'I have a keyfile' button.
6. It it not possible to select the 'I have a keyfile' button without using the mouse at this point.

## Context

While I am perfectly capable of swapping to a mouse to select the keyfile, the process outlined in the 'Expected Behaviour' section above used to function until the new UI came in. It would be nice to have the same functionality back. This might also affect users who actually cannot use a standard mouse for whatever reason.

[NOTE]:
KeePassXC - Version 2.7.7
Revision: 68e2dd8

Qt 5.15.11
Debugging mode is disabled.

Operating system: Windows 11 Version 2009
CPU architecture: x86_64
Kernel: winnt 10.0.22631

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock

Cryptographic libraries:
- Botan 3.1.1

[NOTE]:
Operating System: Windows 11 [Version 23H2 (0S Build 22631.3296)]

**Comment by droidmonkey** [comment]
You just need to press tab one more time, ie, press it twice when you are on the refresh hardware button to move to the key file link. There is nothing we can do about this, it is a Qt quirk. The focus will also not move backwards once on the key file link. I checked with the debugger and everything is as is should be code wise and operationally.

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/9e17ee26-3ef6-4897-a743-5215df9c5bbf)

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/68652888-3b67-4293-8fe9-eedcc8bdb940)

**Comment by Renegade-Master** [comment]
Hello!

Thank you for getting back to me.
I am not sure you understand the issue however.

Yes, it is possible to "tab to the Keyfile button". However, it is not selectable.
Pressing <kbd>Space</kbd> does nothing. Pressing <kbd>Enter</kbd> attempts to open the Database without having selected a Keyfile which results in a failed opening.

The only way to actually select the Keyfile button is with the mouse.

Unless I am missing something.

**Comment by droidmonkey** [comment]
I was able to "press" it with space bar on windows.

**Comment by JohnLGalt** [comment]
I tried (though I'm using a the current snapshot build) and the spacebar would not do anything when the link for 'I have a key file' was definitely highlighted here as well.

However, [ENTER] opened the file dialog box. So, if space doesn't work, try Enter.

Windows 11 (OS Build 22631.3527)
KeePassXC 2.8.0 (latest Snapshot)

**Comment by Renegade-Master** [comment]
Well between us that's

1. Works with <kbd>Space</kbd>
2. Doesn't work with <kbd>Space</kbd>, but works with <kbd>Enter</kbd>
3. Doesn't work with <kbd>Space</kbd>, and also doesn't work with <kbd>Enter</kbd>

So I guess we'll put it down to either keyboards, or some other Windows setting causing different behaviours. It definitely used to work for me, so there might be something I have changed on my machine.

Thank you for investigating anyway!

**Comment by JohnLGalt** [comment]
One of those two not working is cause for concern as it is, but neither working is disconcerting - it's how it should, and always has in the past, worked.

Are you running KPXC as admin by chance?

**Comment by Renegade-Master** [comment]
As Admin? No. I generally would not run normal programs as Admin.

The only behaviour that I get when the 'I have a keyfile' highlighted is that either I press <kbd>Space</kbd> and nothing happens, or I press <kbd>Enter</kbd>, and it goes straight to attempting to open the Database without a keyfile resulting in a failed opening.

It is pretty strange alright. But I am willing to go with the explanation that it is something on my end. I just do not know what on my end.

**Comment by droidmonkey** [comment]
Correction to my previous... `space` does not work for me. Enter does work. @Renegade-Master make sure you see the dashed box around the link text before hitting enter.

**Comment by Renegade-Master** [comment]
You're right! It seems that I have to tab around more than I was expecting, but it definitely works.
I was under the impression that I had the button highlighted, but apparently I didn't.

Thank you for the correction also.
